src/Diagnostics.DataProviders/DataProviders/SupportObserverDataProvider.cs

```diff
@@ -118,6 +119,54 @@ public override async Task<dynamic> GetSite(string stampName, string siteName)
             return siteObject;
         }
 
+        public override async Task<string> GetStampName(string subscriptionId, string resourceGroupName, string siteName)
+        {
+            dynamic siteObjects = await GetSite(siteName);
+            JToken obj2 = ((JArray)siteObjects)
+                    .Select(i => (JObject)i)
+                    .FirstOrDefault(
+                        j => (j.ContainsKey("Subscription") && j["Subscription"].ToString() == subscriptionId
+                            && j.ContainsKey("ResourceGroupName") && j["ResourceGroupName"].ToString() == resourceGroupName
+                            && j.ContainsKey("stampName")));
+
+            string stampName = obj2?["stampName"]?.ToString();
+            return stampName;
+        }
+
+        public override async Task<dynamic> GetHostNames(string siteName)
+        {
+            var response = await Get($"sites/{siteName}/hostnames");
+            var hostNames = JsonConvert.DeserializeObject(response);
+            return hostNames;
+        }
+
+        public override async Task<dynamic> GetSitePostBody(string stampName, string siteName)
+        {
+            var response = await Get($"stamps/{stampName}/sites/{siteName}/postbody");
+            dynamic sitePostBody = JsonConvert.DeserializeObject(response);
+
+            if (sitePostBody["HostNames"] == null)
```

Comment:
>if (sitePostBody["HostNames"] == null) [](start = 12, length = 38)

The postBody response will always not have hostnames filled because the API was rushed when I created it lol. We can make that change in the SupportBayApi if you like so that you don't need to make an API call to get hostnames.